src/main/java/org/manifold/compiler/front/ExpressionVisitor.java

```diff
@@ -2,9 +2,20 @@
 
 public interface ExpressionVisitor {
 
-  void visit(FunctionInvocationExpression functionInvocationExpression);
-  void visit(LiteralExpression literalExpression);
-  void visit(VariableAssignmentExpression variableAssignmentExpression);
-  void visit(VariableReferenceExpression variableReferenceExpression);
+  void visit(FunctionInvocationExpression functionInvocationExpression)
+    throws Exception;
+  void visit(LiteralExpression literalExpression) throws Exception;
+  void visit(VariableAssignmentExpression variableAssignmentExpression) 
+      throws Exception;
+  void visit(VariableReferenceExpression variableReferenceExpression) 
+      throws Exception;
+  void visit(PrimitivePortDefinitionExpression
+      primitivePortDefinitionExpression) throws Exception;
+  void visit(PrimitiveNodeDefinitionExpression
+      primitiveNodeDefinitionExpression) throws Exception;
+  void visit(TupleTypeValueExpression tupleTypeValueExpression) 
+      throws Exception;
+  void visit(FunctionTypeValueExpression functionTypeValueExpression) 
+      throws Exception;
```

Comment:
If by "separate config file" you would allow "a library that is imported", then yes, that is definitely the intent of this design. (Right now everything has to be in one compilation unit, which is why you're seeing these definitions where they are.)